include/seqan3/alphabet/all.hpp

```diff
@@ -39,9 +39,121 @@
  *
  * \defgroup alphabet Alphabet
  *
- * The alphabet module contains different biological alphabets and related functionality.
+ * ## Introduction
+ *
+ * Alphabets are a core component in SeqAn, they enable us to represent the smallest
+ * unit of biological sequence data, e.g. a nucleotide or an amino acid.
+ *
+ * In theory, these could just be represented as a `char` and this is how many people
+ * perceive them, but it makes sense to use a smaller, stricter and well-defined alphabet
+ * in almost all cases, because:
+ *
+ *   * Most of our sequences alphabets are actually smaller and can possibly be **represented by less bits**
+ * than a `char`, e.g. a `char` always uses 8bits of memory, a DNA character *could* be represented by 2bits
+ * if you limit it to the characters 'A', 'C', 'G', 'T';
+ *   * you may want to prevent the user from selecting **letters that aren't part of that alphabet** or converting
+ * those to a special "unknown"-character, e.g. 'J' is not part of any nucleotide definition so people shouldn't
+ * be able to select 'J' and in case you include 'N' in your nucleotide definition you might want 'J' to be
+ * converted to 'N';
+ *   * from a programmer's point of view we are also often interested in the **rank of a letter** inside its
```

Comment:
This is the only reasonable point why we would do this, isn't it.
So maybe this should be stressed more as the motivation for the alphabets.